src/distr/uniform_int.rs

```diff
@@ -402,6 +402,7 @@ uniform_simd_int_impl! { (u8, i8), (u16, i16), (u32, i32), (u64, i64) }
 /// this implementation will use 32-bit sampling when possible.
 #[cfg(any(target_pointer_width = "32", target_pointer_width = "64"))]
 #[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub struct UniformUsize {
```

Comment:
Concern: this isn't portable across pointer widths.

I think it would be sufficient to handle this in the deserializer:

- Default `mode64` to false on 64-bit
- Fail (probably panic) if `mode64 == true` on 32-bit

Reply:
Yes.

Reply:
How about for now we allow serialization for both, and deserialization only for 64-bit:

```
#[cfg_attr(all(feature = "serde"), derive(Serialize))]
#[cfg_attr(
    all(feature = "serde", target_pointer_width = "64"),
    derive(Deserialize)
)]
```

If somebody needs the 32-bit deserialization, they can implement a custom deserializer that on 32-bit:
- panics when `mode64 != false` but otherwise discards it
- panics when the values of the `usize` fields are larger than `u32::MAX`.